src/worker.cpp

```diff
@@ -1217,4 +1218,42 @@ void Worker::collectFrusFromJson()
         }}.detach();
     }
 }
+
+void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
+{
+    try
+    {
+        if (auto l_vpdMap = std::get_if<types::IPZVpdMap>(&io_srcVpdMap))
+        {
+            const std::string& l_imKwdValue = getIMValue(*l_vpdMap);
+            std::unordered_set<std::string> l_imKeys = {"50001001", "50001000",
+                                                        "50001002", "50003000"};
+
+            if (l_imKeys.find(l_imKwdValue) != l_imKeys.end())
+            {
+                BackupAndRestore l_backupAndRestoreObj(m_parsedJson);
+                auto [l_srcVpdVariant, l_dstVpdVariant] =
```

Comment:
Can't we take the return value directly in the passed input parameter? Anyways we are not using destination map.

Reply:
I thought about it, 
what if l_srcVpdVariant value is empty in any wrong scenario, so added below check before updating io_srcVpdMap .

Reply:
Although, I don't see any such possibility, but I am ok with this as well.